jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/IRabitTracker.java

```diff
@@ -0,0 +1,20 @@
+package ml.dmlc.xgboost4j.java;
+
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Interface for Rabit tracker implementations with three public methods:
+ *
+ *  - start(): Start the rabit tracker awaiting for worker connections.
+ *  - getWorkerEnvs(): Return the environment variables needed to initialize Rabit clients.
+ *  - waitFor(): Wait for the task execution by the worker nodes.
+ */
+public interface IRabitTracker {
+  Map<String, String> getWorkerEnvs();
+  boolean start();
+  // for Scala-implementation compatibility
+  boolean start(long timeout, TimeUnit unit);
+  int waitFor();
+  int waitFor(long timeout, TimeUnit unit);
```

Comment:
I see what you mean, if you didn't apply any `action` before XGBoost.train() it may trigger the time-consuming I/O operations.

When using with DataFrame, it might be a rare case. IIRC,  many feature transformers will trigger Spark Jobs 

then this configuration becomes a bit tricky, it highly depends on I/O speed of the users' machines. I will think about it